test/e2e/app-dir/navigation/navigation.test.ts

```diff
@@ -41,18 +44,17 @@ createNextDescribe(
         expect(await browser.elementById('name').text()).toBe('名')
         await browser.elementById('link').click()
 
-        await check(async () => {
-          return requests.some((requestPair) => {
-            const [pathname, ok, headers] = requestPair
-            return (
-              pathname === '/' &&
-              ok &&
-              headers['content-type'] === 'text/x-component'
-            )
-          })
-            ? 'success'
-            : JSON.stringify(requests)
-        }, 'success')
+        await retry(() =>
+          expect(requests).toContainEqual(
+            expect.arrayContaining([
+              '/',
+              true,
+              expect.objectContaining({
+                'content-type': 'text/x-component',
+              }),
+            ])
+          )
```

Comment:
nit: I found the previous version of this more readable when named variables were given to each of the things we were asserting